azdev/operations/pypi.py

```diff
@@ -261,36 +267,109 @@ def verify_versions(modules=None):
 
     results = {}
     for mod, mod_path in modules:
-        if not mod.startswith(COMMAND_MODULE_PREFIX):
+        if not mod.startswith(COMMAND_MODULE_PREFIX) and mod != 'azure-cli':
             mod = '{}{}'.format(COMMAND_MODULE_PREFIX, mod)
         # currently, this is not published
         if mod == 'azure-cli-testsdk':
             continue
-        results.update(_check_module(temp_dir, mod, mod_path))
+        results.update(_compare_module_against_pypi(temp_dir, mod, mod_path))
 
     shutil.rmtree(temp_dir)
     os.chdir(original_cwd)
 
+    results = _check_setup_py(results, no_update)
+
     logger.info('Module'.ljust(40) + 'Local Version'.rjust(20) + 'Public Version'.rjust(20))  # pylint: disable=logging-not-lazy
     for mod, data in results.items():
         logger.info(mod.ljust(40) + data['local_version'].rjust(20) + data['public_version'].rjust(20))
 
-    failed_mods = {k: v for k, v in results.items() if v['status'] != 'OK'}
+    bump_mods = {k: v for k, v in results.items() if v['status'] == 'BUMP'}
+    mismatch_mods = {k: v for k, v in results.items() if v['status'] == 'MISMATCH'}
     subheading('RESULTS')
-    if failed_mods:
+    if bump_mods:
         logger.error('The following modules need their versions bumped. '
-                     'Scroll up for details: %s', ', '.join(failed_mods.keys()))
+                     'Scroll up for details: %s', ', '.join(bump_mods.keys()))
         logger.warning('\nNote that before changing versions, you should consider '
                        'running `git clean` to remove untracked files from your repo. '
                        'Files that were once tracked but removed from the source may '
                        'still be on your machine, resuling in false positives.')
+    elif mismatch_mods and no_update:
+        logger.error('The following modules have a mismatch between the module version '
+                     'and the version in azure-cli\'s setup.py file. '
+                     'Scroll up for details: %s', ', '.join(mismatch_mods.keys()))
     else:
         display('OK!')
 
 
+def _check_setup_py(results, no_update):
+    # only audit or update setup.py when all modules are being considered
+    # otherwise, edge cases could arise
+    if no_update is None:
+        return results
+
+    # retrieve current versions from azure-cli's setup.py file
+    azure_cli_path = get_path_table(include_only='azure-cli')['core']['azure-cli']
+    azure_cli_setup_path = find_files(azure_cli_path, SETUP_PY_NAME)[0]
+    with open(azure_cli_setup_path, 'r') as f:
+        setup_py_version_regex = re.compile(r"'([^'=]*)==([\d.]*)'")
```

Comment:
Another small thing with regex, I always prefer named groups. IMHO, they're more readable, and more resilient to changes to the pattern itself later.